src/zip.cpp

```diff
@@ -131,26 +128,26 @@ namespace zip
     pfileIn = fopen(zi.filename.c_str(), "rb"); // open ZIP file for reading
     if (pfileIn == nullptr) {
       LOG_ERROR("Couldn't open zip file for reading: " << zi.filename);
+      fclose(*pfileOut);
       return ERR_FILE_UNZIP_FAILED; // couldn't open input file
     }
+    auto closure  = [&]() { fclose(pfileIn); };
```

Comment:
Maybe the closure and scope_exit variable should be named something like inFileCloser, making it clearer if we do it for both in and out. Alternatively a single closure closes both and we keep a fclose for pFileOut

Reply:
Makes sense